akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -1339,11 +1358,26 @@ private[cluster] final class FirstSeedNodeProcess(seedNodes: immutable.IndexedSe
         context.parent ! JoinTo(selfAddress)
         context.stop(self)
       }
-    case InitJoinAck(address) ⇒
+
+    case InitJoinAck(address, true) ⇒
       logInfo("Received InitJoinAck message from [{}] to [{}]", sender(), selfAddress)
       // first InitJoinAck reply, join existing cluster
       context.parent ! JoinTo(address)
       context.stop(self)
+
+    case InitJoinAck(address, false) ⇒
+      // first InitJoinAck reply, but incompatible
+      if (!EnforceConfigCompatCheck) {
+        // only join if set to ignore config validation
+        logInfo("Received InitJoinAck message from [{}] to [{}]", sender(), selfAddress)
+        log.warning("Joining cluster without configuration compatibility check.")
+        context.parent ! JoinTo(address)
+        context.stop(self)
+      } else {
+        log.error("Couldn't join seed nodes because of incompatible configurations. Check cluster logs for details!")
+        context.stop(self)
```

Comment:
Thinking about what we should do here. Do we want it to continue retrying which would only be successful when all old incompatible nodes have been shutdown, or shall we
```
CoordinatedShutdown(context.system).run()
```

since `shutdown-after-unsuccessful-join-seed-nodes` is by default `off` that will not stop things.

Reply:
What happens if we have a 3 nodes cluster and we bring 3 new nodes with incompatible config and then remove the 3 old nodes?

The 3 new nodes will keep trying, I understand, but will they eventually form a cluster when the 3 old nodes get out of the way?

I'm assuming here that the addresses are being somehow resolved and the 3 new nodes can see each other.

Reply:
yes, given that both old and new nodes are included in the `seed-nodes` list they will continue trying. They will not succeed joining the old nodes but if all old have been removed and one of the new is listed as first seed-node it can join itself and thereby form a new cluster when the old have been removed.

However, that might be too complicated. Perhaps better to just shutdown and say that if you want to start a new cluster you should first stop the old nodes.